packages/Interface/src/DTK_C_API.h

```diff
@@ -64,6 +64,9 @@ extern bool DTK_is_initialized();
 /*! Finalize DTK */
 extern void DTK_finalize();
 
+/*! Get error message */
+extern int DTK_error( int err, char *msg );
```

Comment:
Not sure I understand you, but yes, this should be called as soon as we understand that the call failed.